content/version_control/VCS.tex

```diff
@@ -0,0 +1,174 @@
+  \inst{1}
+  Hochschule Bonn-Rhein-Sieg
+
+}
+
+\begin{document}
+
+\begin{frame}
+  \titlepage
+\end{frame}
+
+\begin{frame}{Outline}
+  \tableofcontents
+\end{frame}
+
+\begin{frame}{What is it?}
+\section{What is it?}
+  \begin{itemize}
+  \item {
+    A method to manage file changes over time. These files are stored in a central location.
+  }
+  \item {
+    A backup when things go wrong.
+  }
+  \item{A software that helps team members collaborate with minimum disruptions.}
+  \end{itemize}
+\end{frame}
+
+\begin{frame}{Why should we use it ?}
+\section{Why should we use it ?}
+  \begin{itemize}
+  \item {
+    A complete long-term change history of every file.
+  }
+  \item {   
+    Branching and merging
+  }
+  \item {   
+    Ability to trace each change made
+  }
+  \item {   
+	Allows multiple developers to work simultaneously
+  }
+  \end{itemize}
+\end{frame}
+
+\begin{frame}{Types of VCS}{Centralized VCS}
+\section{Types of VCS}
+\subsection{Centralized VCS}
+  \begin{columns}
+\column{0.5\textwidth}
+\begin{itemize}
+  \item {
+    Centralized VCS keeps the history of changes on a central server.
+  }
+  \item {
+    Designed with the intent that there is only one true source  
+  }
+  \item {
+    If you want to make a copy of your data, you have to copy/paste it, literally.  
+  }
+  \item{
+	Sourceforge.net uses this type of versioning.  
+  }
+  \item{
+      \textbf{Example:} SVN (Subversion)
+  }  
+  \end{itemize}
+\column{0.5\textwidth}
+\begin{figure}
+	 \includegraphics[width=.9\textwidth]{images/cvcs}
+	 \caption{Centralized Version System \cite{website}}
+ \end{figure}
+\end{columns}
+\end{frame}	
+
+\begin{frame}{Types of VCS}{Decentralized VCS}
+\subsection{Distributed VCS}
+\begin{columns}
+\column{0.5\textwidth}
+\begin{itemize}
+		  \item {
+		    In distributed VCS, everyone has a local copy of the entire work’s history
+		  }
+		  \item {
+		    Each repository is as good as the other
+		  }
+		  \item{
+			Prone to central server crash
+		  }
+		  \item{
+			Your local copy is a repository, and you can commit to it and get all benefits of source control - Offline Source Control.
+		  }	  
+		  \item{
+			Mozilla Firefox uses this kind of versioning.  
+		  }
+		  \item{
+           \textbf{ Example: }Git		  
+		  }
+	  \end{itemize}
+\column{0.5\textwidth}
+\begin{figure}
+	 \includegraphics[width=.9\textwidth]{images/dvcs}
+	 \caption{Distributed Version System \cite{website}}
+ \end{figure}
+\end{columns}
+\end{frame}	
+
+\begin{frame}[allowframebreaks]{Types of VCS}{Git vs SVN}
+\subsection{Git vs SVN}
+
+\textbf{Pros of using SVN}
+\begin{itemize}
+    \item Subversion has better GUIs than Git
+    \item Working through versions in Git is tougher than SVN. Git uses SHA-1 hashes while SVN uses sequential revision numbers. 
+\end{itemize}
+\textbf{Pros of using Git}
+\begin{itemize}
+    \item Git is much faster than SVN as they do not have to keep communicating with a central server when changes are made.
```

Comment:
I think one of the biggest advantages of git is merging, have you looked into this?

Reply:
I will explain that point with the 3rd pro of Git

Reply:
I think it's better if the idea of why is it easier to merge using git than svn should be mentioned here, and the practical aspects of merging in the git session